runtimes/runtimes/base-runtime.ts

```diff
@@ -116,8 +116,10 @@ export const baseRuntime = (connections: { reader: MessageReader; writer: Messag
             copyFile: (_src, _dest, _options?) => Promise.resolve(),
             exists: _path => Promise.resolve(false),
             getFileSize: _path => Promise.resolve({ size: 0 }),
-            getServerDataDirPath: _serverName => '',
+            getServerDataDirPath: serverName =>
+                lspRouter.clientInitializeParams!.initializationOptions?.aws?.clientDataFolder ?? `/${serverName}`,
```

Comment:
since this already handles falsy, can avoid "!"

```suggestion
                lspRouter.clientInitializeParams?.initializationOptions?.aws?.clientDataFolder ?? `/${serverName}`,
```